### PR TITLE
[FIX] stock: wrong check to allow update UoM

### PR DESCRIPTION
**Pull request by simongoffin**
- If at least one unit of a category was already used in a stock.move.line, it was not possible to modify a not used unit in this category.
- The check was not perform on all stock.move.line (user sudo()).

opw:2006965

### CODE REVIEW

**Comment by fmdl** [comment]
any news ?

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -597,23 +598,41 @@ class ProductUoM(models.Model):
...
+
+        if changed_ref or changed_mult:
+            dom_list = []
+            if changed_ref:
+                dom_list.append([('product_uom_id.category_id', 'in', changed_ref.mapped('category_id').ids)])
+            if changed_mult:
+                dom_list.append([('product_uom_id', 'in', changed_mult.ids)])
+
+            domain = [('state', '!=', 'cancel')]
+            domain.extend(expression.OR(dom_list))
```
Comment (amoyaux) [inline addons/stock/models/product.py]: AND

File: addons/stock/models/product.py [hunk 2]
```diff
@@ -597,23 +598,41 @@ class ProductUoM(models.Model):
...
+            uom_used = self.env['stock.move.line'].sudo().search(domain, limit=1)
```
Comment (amoyaux) [inline addons/stock/models/product.py]: search_count

**Comment by pedrobaeza** [comment]
Coming to this through codetriage.com. Is this still relevant?

**Comment by C3POdoo** [comment]
Dear @simongoffin,

Thank you for your contribution but the version 11.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**